src/app/main/main.component.js

```diff
@@ -40,6 +49,21 @@ function routingConfig($stateProvider, $locationProvider, $urlRouterProvider){
       url: "/checkout.html",
       template: '<checkout></checkout>'
     })
+    .state('profile', {
+      url: "/profile",
+      abstract: true,
+      template: '<profile></profile>',
+      onEnter: (sessionService,sessionModalService,$state) => {
+        if(sessionService.getRole() != 'REGISTERED') {
```

Comment:
I've written a sessionEnforcerService to facilitate session enforcement and expiration in components. You can see an example of it used in [checkout](https://github.com/CruGlobal/give-web/blob/master/src/app/checkout/checkout.component.js#L39)